libs/core/src/lib/message-page/message-page.component.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ChangeDetectorRef,
+    ElementRef,
+    Input,
+    OnChanges,
+    ViewEncapsulation,
+    OnInit
+} from '@angular/core';
+import { applyCssClass, CssClassBuilder } from '../utils/public_api';
+
+export type MessagePageType = '' | 'filter' | 'search' | 'no-items' | 'error';
+
+@Component({
+    // tslint:disable-next-line:component-selector
+    selector: 'fd-message-page',
+    templateUrl: './message-page.component.html',
+    styleUrls: ['./message-page.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MessagePageComponent implements OnChanges, OnInit, CssClassBuilder {
+    /** User's custom class */
+    @Input()
+    class: string;
+
+    /** 
+     * The type of the Message Page.
+     * Options include: filter, search, no-items, error and default.
+     * Leave empty for default.
+     */
+    @Input()
+    type: MessagePageType;
+
+     /** 
+     * Whether the Message Page has an icon.
+     * Set to true by default.
+     * It's recommended to always show an icon with the Message Page. 
+     */
+    @Input()
+    hasIcon = true;
+
+    /** 
+     * The icon of the Message Page.
+     * The default icon is .sap-icon--documents. 
+     */
+    @Input()
+    set glyph(messagePageGlyph: string) {
+        if (messagePageGlyph) {
+            this._glyph = messagePageGlyph;
+            this._getMessagePageIcon();
+            this._changeDetectorRef.detectChanges();
+        }
+        
+    }
+
+    /** 
+     * Getter method for the glyph property
+     */
+    get glyph(): string {
+        return this._glyph;
+    }
+
+    /**
+     * @hidden
+     * The icon of the Message Page.
+     */
+    private _glyph = 'documents';
+
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef, private _changeDetectorRef: ChangeDetectorRef) {}
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this.buildComponentCssClass();
+    }
+
+    @applyCssClass
+    /** CssClassBuilder interface implementation
+     * function is responsible for order which css classes are applied
+     */
+    buildComponentCssClass(): string[] {
+        return [
+            'fd-message-page',
+            this.class
+        ];
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef {
+        return this._elementRef;
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.buildComponentCssClass();
+        this._getMessagePageIcon();
+        this._changeDetectorRef.detectChanges();
```

Comment:
I also think this `detectChanges` is unnecessary

Reply:
removed